celery_pubsub/pubsub.py

```diff
@@ -1,61 +1,37 @@
+import re
 import celery
 
-__all__ = ['publish', 'publish_now', 'subscribe', 'unsubscribe', ]
-
 
 class PubSubManager(object):
+
     def __init__(self):
-        super(PubSubManager, self).__init__()
-        self.subscribed = []
         self.jobs = {}
 
     def publish(self, topic, *args, **kwargs):
-        result = self.get_jobs(topic).delay(*args, **kwargs)
-        return result
+        group = []
+        for (regex, task) in self.jobs.values():
+            if regex.match(topic):
+                group.append(task)
+        out = celery.group(group)
+        return out
 
     def publish_now(self, topic, *args, **kwargs):
-        result = self.get_jobs(topic).apply(args=args, kwargs=kwargs)
+        # REVIEW: useless shortcut?
+        result = self.publish(topic, *args, **kwargs).apply(args=args, kwargs=kwargs)
         return result
 
     def subscribe(self, topic, task):
-        regex = self._topic_to_re(topic)
-        if (topic, regex, task) not in self.subscribed:
-            self.subscribed.append((topic, regex, task))
-            self.jobs = {}
-
-    def unsubscribe(self, topic, task):
-        found = -1
-        for idx, triplet in enumerate(self.subscribed):
-            if (triplet[0], triplet[2]) == (topic, task):
-                found = idx
-                break
-        if not found == -1:
-            del self.subscribed[found]
-            self.jobs = {}
```

Comment:
This is a case for [`for .. break ... else`](https://docs.python.org/3/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loops)

Reply:
Isn't that the opposite?
The `for else` should be executed if there was no `break`. But I'm executing it when there is a `break`. Plus, I needed the index value, so the extra variable was already needed.